### PR TITLE
Replace BlockSolutionAccessComponent with an attempt event

### PR DESCRIPTION
**Pull request by Tayrtahn**
## About the PR

BlockSolutionAccessComponent has been replaced by an access attempt event.

## Why / Balance

Discord discussion rightfully pointed out that the current implementation is clumsy and excessive. This change is a bit nicer.

## Technical details

Removed BlockSolutionAccessComponent.
Moved spilling-while-worn functionality to its own component: `SpillWhenWornComponent`.
Added an event for attempting to access a named solution on an entity. SpillWhenWorn cancels the attempt if the entity is being worn.

## Media

Code
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

### CODE REVIEW

File: Content.Server/Fluids/EntitySystems/PuddleSystem.Spillable.cs [hunk 1]
```diff
@@ -117,7 +117,9 @@ private void OnGotEquipped(Entity<SpillableComponent> entity, ref GotEquippedEve
             return;
 
         // block access to the solution while worn
-        AddComp<BlockSolutionAccessComponent>(entity);
+        var blockerComp = AddComp<BlockSolutionAccessComponent>(entity);
+        blockerComp.Solution = entity.Comp.SolutionName;
+        Dirty(entity, blockerComp);
```
Comment (metalgearsloth) [inline Content.Server/Fluids/EntitySystems/PuddleSystem.Spillable.cs:123]: Isn't this going to get bulldozed if somehow they wear 2 things with this?

Comment (deltanedas) [reply Content.Server/Fluids/EntitySystems/PuddleSystem.Spillable.cs:123]: no it will throw because AddComp

Comment (Tayrtahn) [reply Content.Server/Fluids/EntitySystems/PuddleSystem.Spillable.cs:123]: Argh, good points.

It's easy enough to switch to EnsureComp, but then there's the bulldozing issue. I'll swap the single value for a set.

Comment (Tayrtahn) [reply Content.Server/Fluids/EntitySystems/PuddleSystem.Spillable.cs:123]: Wait, no. The component is added to the clothing entity and blocks access to its own solutions, not the wearer's.  Someone wearing multiple things with Spillable won't cause any problems, they'll just each block their own internal solution.

Comment (Tayrtahn) [reply Content.Server/Fluids/EntitySystems/PuddleSystem.Spillable.cs:123]: I suppose there could be an issue if multiple systems tried to use BlockSolutionAccess on one entity. Spillable is the only thing that uses it now, but maybe something else would in the future. Hm.

Comment (ShadowCommander) [reply Content.Server/Fluids/EntitySystems/PuddleSystem.Spillable.cs:123]: Does it work if you make blockerComp.Solution a list?

**Review by ShadowCommander — APPROVED** [review]

**Pull request merged**